### ISSUE TITLE
qase-pytest: Run not getting completed when running pytests in parallel

### ISSUE THREAD

**Issue by KrishnaYadav2102** [qase-tms/qase-python#33]
Used the following command:
`pytest --workers auto --qase --qase-api-token=<token> --qase-project=<code> --qase-testplan=<plan> tests/`

I was trying to run pytest in parallel & got the following exception:
```
..INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pytest_parallel/__init__.py", line 93, in run
INTERNALERROR>     run_test(self.session, item, None)
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pytest_parallel/__init__.py", line 54, in run_test
INTERNALERROR>     item.ihook.pytest_runtest_protocol(item=item, nextitem=nextitem)
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pluggy/_hooks.py", line 265, in __call__
INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pluggy/_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pluggy/_callers.py", line 60, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pluggy/_result.py", line 60, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pluggy/_callers.py", line 34, in _multicall
INTERNALERROR>     next(gen)  # first yield
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/qaseio/pytest/plugin.py", line 296, in pytest_runtest_protocol
INTERNALERROR>     self.start_pytest_item(item)
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/qaseio/pytest/plugin.py", line 337, in start_pytest_item
INTERNALERROR>     result = self.client.results.create(
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/qaseio/client/services/results.py", line 40, in create
INTERNALERROR>     self.s.post(
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/requests/sessions.py", line 635, in post
INTERNALERROR>     return self.request("POST", url, data=data, json=json, **kwargs)
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/apitist/requests.py", line 252, in request
INTERNALERROR>     resp = ApitistResponse(self.send(prep, **send_kwargs))
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/requests/sessions.py", line 701, in send
INTERNALERROR>     r = adapter.send(request, **kwargs)
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/requests/adapters.py", line 563, in send
INTERNALERROR>     raise SSLError(e, request=request)
INTERNALERROR> requests.exceptions.SSLError: None: Max retries exceeded with url: /v1/result/WEBEDI/130 (Caused by None)
INTERNALERROR> 
INTERNALERROR> The above exception was the direct cause of the following exception:
INTERNALERROR> 
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/_pytest/main.py", line 268, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/_pytest/main.py", line 322, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pluggy/_hooks.py", line 265, in __call__
INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pluggy/_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pluggy/_callers.py", line 60, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pluggy/_result.py", line 60, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pluggy/_callers.py", line 39, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/krishnayadav/Krishna/repositorties/fastfood/venv38/lib/python3.8/site-packages/pytest_parallel/__init__.py", line 334, in pytest_runtestloop
INTERNALERROR>     six.raise_from(exc, err[1])
INTERNALERROR>   File "<string>", line 3, in raise_from
INTERNALERROR> RuntimeError: pytest-parallel got 8 errors, raising the first from Thread-2.
```

**Comment by KrishnaYadav2102** [comment]
Is there any plan to fix this issue?

**Comment by KrishnaYadav2102** [comment]
I have raised the following ticket:
https://qase.canny.io/feature-requests/p/support-for-pytest-parallel-execution

I need some votes to get this on QASE roadmaps.

**Comment by n3r** [comment]
Will be fixed in v4.0.0.

**Comment by KrishnaYadav2102** [comment]
Awesome
Thank you Nikita

When is release v4.0.0 is planned?

**Comment by n3r** [comment]
Fixed in qase-pytest-4.0.0 #64

If you want to parallelise test execution, you can use `[pytest-xdist](https://pypi.org/project/pytest-xdist/)`. Reporter has a native support for it.
